usercp.php

```diff
@@ -57,67 +57,20 @@
 $plugins->run_hooks("usercp_start");
 if($mybb->input['action'] == "do_editsig" && $mybb->request_method == "post")
 {
-	$parser_options = array(
-		'allow_html' => $mybb->settings['sightml'],
-		'filter_badwords' => 1,
-		'allow_mycode' => $mybb->settings['sigmycode'],
-		'allow_smilies' => $mybb->settings['sigsmilies'],
-		'allow_imgcode' => $mybb->settings['sigimgcode'],
-		"filter_badwords" => 1
+	require_once MYBB_ROOT."inc/datahandlers/user.php";
+	$userhandler = new UserDataHandler();
+	
+	$data = array(
+		'uid' => $mybb->user['uid'],
+		'showimages' => $mybb->user['showimages'],
+		'signature' => $mybb->get_input('signature'),
 	);
 
-	if($mybb->user['showimages'] != 1 && $mybb->user['uid'] != 0)
-	{
-		$parser_options['allow_imgcode'] = 0;
-	}
+	$userhandler->set_data($data);
 
-	$parsed_sig = $parser->parse_message($mybb->get_input('signature'), $parser_options);
-	if((($mybb->settings['sigimgcode'] == 0 && $mybb->settings['sigsmilies'] != 1) &&
-		substr_count($parsed_sig, "<img") > 0) ||
-		(($mybb->settings['sigimgcode'] == 1 || $mybb->settings['sigsmilies'] == 1) &&
-		substr_count($parsed_sig, "<img") > $mybb->settings['maxsigimages'])
-	)
-	{
-		if($mybb->settings['sigimgcode'] == 1)
-		{
-			$imgsallowed = $mybb->settings['maxsigimages'];
-		}
-		else
-		{
-			$imgsallowed = 0;
-		}
-		$lang->too_many_sig_images2 = $lang->sprintf($lang->too_many_sig_images2, $imgsallowed);
-		$error = inline_error($lang->too_many_sig_images." ".$lang->too_many_sig_images2);
-		$mybb->input['preview'] = 1;
-	}
-	else if($mybb->settings['siglength'] > 0)
-	{
-		if($mybb->settings['sigcountmycode'] == 0)
-		{
-			$parsed_sig = $parser->text_parse_message($mybb->get_input('signature'));
-		}
-		else
-		{
-			$parsed_sig = $mybb->get_input('signature');
-		}
-		$parsed_sig = preg_replace("#\s#", "", $parsed_sig);
-		$sig_length = my_strlen($parsed_sig);
-		if($sig_length > $mybb->settings['siglength'])
-		{
-			$lang->sig_too_long = $lang->sprintf($lang->sig_too_long, $mybb->settings['siglength']);
-			if($sig_length - $mybb->settings['siglength'] > 1)
-			{
-				$lang->sig_too_long .= $lang->sprintf($lang->sig_remove_chars_plural, $sig_length-$mybb->settings['siglength']);
-			}
-			else
-			{
-				$lang->sig_too_long .= $lang->sig_remove_chars_singular;
-			}
-			$error = inline_error($lang->sig_too_long);
-		}
-	}
-	if(isset($error) || !empty($mybb->input['preview']))
+	if (!$userhandler->verify_signature())
```

Comment:
No space before `(`. There also might be problem with this conditional (I recall userhandler error from another PRs), I'll test it later. For now that's all I found by reading the code.